### DISCUSSION TITLE
Main different between GitHub Workflows and Environment

### DISCUSSION THREAD

**Discussion by DavideC03** [github/india#21] [Satellite India 2021 Q&A] [answered]
Can you explain me that's the main differences between Workflows and Environment?

Thanks,
DavideC03

**Comment by chrispat** [comment, marked as answer]
A Workflow how you define what actions you want to run in response to a given event like 'push' or 'pull_request'.  An environment can be referenced by a workflow to allow it to access protected secrets as well as give you a deployment history of your app.

**Reply by DavideC03** [reply]
Thanks

**Comment by FRTMentor** [comment]
A Workflow how you define what actions you want to run in response to a given event like 'push' or 'pull_request'. An environment can be referenced by a workflow to allow it to access protected secrets as well as give you a deployment history of your app.
